### PR TITLE
Added some simple small examples to files promise type docs

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

File: reference/promise-types/files.markdown [hunk 1]
```diff
@@ -7,7 +7,78 @@ tags: [reference, bundle agent, files, promises, files promises, promise types]
...
+    "/tmp/hello"
+      content => "Hello, CFEngine";
+}
+```
+
+Before making changes to the system, CFEngine will check what the current state is, and whether it matches the desired state.
+In this case, CFEngine will create and edit the file only if necessary, if `/tmp/hello` already exists with the correct content, no changes are made.
+
+**Tip:** In this and other examples, we use the `__main__` bundle.
+This allows you to easily run a policy file directly.
```
Comment (larsewi) [inline reference/promise-types/files.markdown:27]:
```suggestion
This bundle is evaluated when (and only when) running the policy file directly.
```

File: reference/promise-types/files.markdown [hunk 2]
```diff
@@ -7,7 +7,78 @@ tags: [reference, bundle agent, files, promises, files promises, promise types]
...
+bundle agent __main__
+{
+  files:
+    "/tmp/hello"
+      content => "Hello, CFEngine",
+      if => fileexists("/tmp/hello");
+}
+```
+
+In the example above, `fileexists()` is a built in function call.
```
Comment (larsewi) [inline reference/promise-types/files.markdown:44]:
```suggestion
In the example above, `fileexists()` is a built-in function call.
```

**Review by larsewi — APPROVED** [review]
Few suggestions. Otherwise a great improvement in terms of ease-of-use. Thanks!

File: reference/promise-types/files.markdown (hunk 1 above)
Comment (olehermanse) [reply reference/promise-types/files.markdown:27]: No, my example is easier / more focused on the user, IMO.

File: reference/promise-types/files.markdown [hunk 3]
```diff
@@ -7,7 +7,78 @@ tags: [reference, bundle agent, files, promises, files promises, promise types]
...
+bundle agent __main__
+{
+  files:
+    ubuntu_18|ubuntu_20::
+      "/tmp/hello"
+        create => "true";
+}
+```
+
+The line with `ubuntu_18|ubuntu_20::` is called a class guard - it means the promise will only be evaluated if the `ubuntu_18` or `ubuntu_20` class is defined.
```
Comment (nickanderson) [inline reference/promise-types/files.markdown:59]: maybe link to glossary, also called class expression

File: reference/promise-types/files.markdown [hunk 4]
```diff
@@ -7,7 +7,78 @@ tags: [reference, bundle agent, files, promises, files promises, promise types]
...
+There are many others available to use, see: [Functions][Functions].
+
+### On specific platforms, ensure a file exists by creating it if necessary
+
+```cf3
+bundle agent __main__
+{
+  files:
+    ubuntu_18|ubuntu_20::
+      "/tmp/hello"
```
Comment (nickanderson) [inline reference/promise-types/files.markdown:54]: this one doesn't age well. Suggest create a file for NOT ubuntu_16 or something that will keep the output the same over time.

File: reference/promise-types/files.markdown [hunk 5]
```diff
@@ -7,7 +7,78 @@ tags: [reference, bundle agent, files, promises, files promises, promise types]
...
+
+The with attribute and variable expansion allows you to easily store the output of a function in a temporary variable, which you can then expand inside content, or other strings within the same promise:
+
+```cf3
+bundle agent __main__
+{
+  files:
+    "/tmp/hello"
+      content => "Output of uname: $(with)",
+      with => execresult("uname", "useshell");
```
Comment (nickanderson) [inline reference/promise-types/files.markdown:75]: This doesn't age well, uname output will change necessitating changes to the example. Suggest perhaps something like `stat -c "%U" /` which should always return root.

**Review by nickanderson — COMMENTED** [review]
I think the inline examples should be replaced with examples from core/examples (which do double duty as tests).

```
[%CFEngine_include_example(<Filename from core/examples>)%]
```

E.g. with these new examples from this PR: https://github.com/cfengine/core/pull/5046

```
[%CFEngine_include_example(files-content.cf)%]
[%CFEngine_include_example(files-content-if-fileexists.cf)%]
[%CFEngine_include_example(files-content-create-class-expression.cf)%]
[%CFEngine_include_example(files-content-with.cf)%]
```

**Pull request merged**